File: includes/C/C++/jumping-numbers.cpp
#include <bits/stdc++.h> 
using namespace std;

vector<long> v;

void jumpingNumberUsingBFS(int jump, int n) {
 queue<int> q;
 q.push(jump);
 while(!q.empty()){
     jump = q.front();
     q.pop();
     if(jump <= n){
         v.push_back(jump);
         int last_digit = jump % 10;
         if(last_digit == 0) {
             q.push((jump*10)+last_digit+1);
         } else if(last_digit == 9){
             q.push((jump*10)+last_digit-1);
         } else {
             q.push((jump*10)+last_digit-1);
             q.push((jump*10)+last_digit+1);
         }
     }
 }
}

int main(){
    int T;
    cin>>T;
    while(T--){
        int n;
        cin>>n;
        v.push_back(0);
        for(int i=1; (i<=9 && i<=n); i++){
            jumpingNumberUsingBFS(i, n);
        }
        sort(v.begin(), v.end());
        for (const auto &i: v)
          cout << i << ' ';
    }
    return 0;
}

File: includes/JavaScript/jumping-numbers.js
// Input:
// 2
// 10
// 50
// Output:
// 0 1 2 3 4 5 6 7 8 9 10
// 0 1 2 3 4 5 6 7 8 9 10 12 21 23 32 34 43 45
function jumpingNumberUsingBFS(jump, n) {
    var q = [];
    var arr = []
    q.push(jump);
    while (q.length) {
        jump = q[0];
        q.shift();
        if (jump <= n) {
            arr.push(jump);
            let last_digit = jump % 10;
            if (last_digit == 0) {
                q.push((jump * 10) + last_digit + 1);
            } else if (last_digit == 9) {
                q.push((jump * 10) + last_digit - 1);
            } else {
                q.push((jump * 10) + last_digit - 1);
                q.push((jump * 10) + last_digit + 1);
            }
        }
    }
    return arr;
}

function findNumber(){
    // var n = window.promt("enter input")
    var arr = [0];
    for (let i = 1; (i <= 9 && i <= 50); i++) {
        arr = [...arr, ...jumpingNumberUsingBFS(i, 50)];
    }
    console.log(arr.sort(function(a, b){return a - b}));
}
findNumber();